src/core/catalog.js

```diff
@@ -438,13 +438,13 @@ class Catalog {
         return shadow(this, "optionalContentConfig", null);
       }
       const groups = [];
-      const groupRefs = [];
+      const groupRefs = new RefSet();
       // Ensure all the optional content groups are valid.
       for (const groupRef of groupsData) {
-        if (!(groupRef instanceof Ref)) {
+        if (!(groupRef instanceof Ref) || groupRefs.has(groupRef)) {
           continue;
         }
-        groupRefs.push(groupRef);
+        groupRefs.put(groupRef);
         const group = this.xref.fetchIfRef(groupRef);
```

Comment:
This probably doesn't help much overall, but given the checks above aren't we always *guaranteed* to have a `Ref` here?
```suggestion
        const group = this.xref.fetch(groupRef);
```